jobspec/parse.go

```diff
@@ -17,7 +17,7 @@ import (
 	"github.com/mitchellh/mapstructure"
 )
 
-var reDynamicPorts = regexp.MustCompile("^[a-zA-Z0-9_]+$")
+var reDynamicPorts = regexp.MustCompile("^[a-zA-Z0-9_-]+$")
```

Comment:
To have a `-` in a regex, we need to escape it:

```suggestion
var reDynamicPorts = regexp.MustCompile("^[a-zA-Z0-9_\\-]+$")
```